app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -10,6 +10,21 @@ class MetaRPCClient {
     this.requests = new Map();
     this.connectionStream.on('data', this.handleResponse.bind(this));
     this.connectionStream.on('end', this.close.bind(this));
+    this.responseHandled = false;
+  }
+
+  send(id, payload, cb) {
+    this.responseHandled = false;
+    this.requests.set(id, cb);
+    this.connectionStream.write(payload);
+    setTimeout(() => {
+      if (!this.responseHandled) {
+        return cb(new Error('No response from RPC'), null);
+      }
+
+      // needed for linter to pass
+      return true;
+    }, 10000);
```

Comment:
Nit: for readability, I'd suggesting pulling this out into a constant with a nicer name, e.g. `TEN_SECONDS_IN_MILLISECONDS`. Also we can use numeric separators now, e.g. `10_000` to make the literal number easier to read.